### PR TITLE
Create PVC first if it's referenced by PV

### LINKED ISSUE #96658: [integration test] PVC should create before the PV if it's referenced by PV
**What happened**:
Run integration test `BenchmarkPerfScheduling`, for the case which needs PV or PVC,  got the error which info us that PVC is not found, 

```
make test-integration WHAT=./test/integration/scheduler_perf KUBE_TEST_VMODULE="''" KUBE_TEST_ARGS="-alsologtostderr=false -logtostderr=false -run=^$$ -benchtime=1ns -bench=BenchmarkPerfScheduling/SchedulingInTreePVs/500Nodes/500InitPods/1000PodsToSchedule"

E1117 09:25:17.819958   22269 util.go:120] error while getting namespace-1/pvc-1: persistentvolumeclaims "pvc-1" not found
E1117 09:25:17.820652   22269 util.go:120] error while getting namespace-1/pvc-0: persistentvolumeclaims "pvc-0" not found
E1117 09:25:17.821858   22269 util.go:120] error while getting namespace-1/pvc-1: persistentvolumeclaims "pvc-1" not found
```

Furtherly, this will lead a inconsistent PV / PVC status, and fail the whole testcase.
 
**What you expected to happen**:
Run successfully.

**How to reproduce it (as minimally and precisely as possible)**:
```
make test-integration WHAT=./test/integration/scheduler_perf KUBE_TEST_VMODULE="''" KUBE_TEST_ARGS="-alsologtostderr=false -logtostderr=false -run=^$$ -benchtime=1ns -bench=BenchmarkPerfScheduling/SchedulingInTreePVs/500Nodes/500InitPods/1000PodsToSchedule"
```

**Anything else we need to know?**:
The error message is thrown here, 
https://github.com/kubernetes/kubernetes/blob/36d12390a724c59f44787968deca18de92f9181c/test/integration/util/util.go#L115-L134

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/sig scheduling
/sig testing

### PR DESCRIPTION
**Pull request by chendave**
Signed-off-by: Dave Chen <dave.chen@arm.com>

/sig testing
/sig scheduling

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/96658

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by chendave** [comment]
/cc @cofyc
Hi Yecheng, do you have any comment for this change?

File: test/utils/runners.go [hunk 1]
```diff
@@ -1353,6 +1353,21 @@ func CreatePodWithPersistentVolume(client clientset.Interface, namespace string,
...
 		pvc.Name = pvcName
+
+		// Create PVC first as it's referenced by the PV when the `bindVolume` is true.
+		if err := CreatePersistentVolumeClaimWithRetries(client, namespace, pvc); err != nil {
+			lock.Lock()
+			defer lock.Unlock()
+			createError = fmt.Errorf("error creating PVC: %s", err)
+			return
+		}
+		if _, err := client.CoreV1().PersistentVolumeClaims(namespace).UpdateStatus(context.TODO(), pvc, metav1.UpdateOptions{}); err != nil {
```
Comment (oomichi) [inline test/utils/runners.go:1364]: If moving this function call before the original line 1373 `pvc.Status.Phase = v1.ClaimBound`, the pvc status change is not affected on this function. Do I miss something?

**Review by oomichi — COMMENTED** [review]
just one question.

File: test/utils/runners.go (hunk 1 above)
Comment (chendave) [reply test/utils/runners.go]: Nice catch! updated in the new PR.

**Comment by chendave** [comment]
Hi @oomichi, thanks for the review, could you pls take another look?

**Review by oomichi — COMMENTED** [review]
Thanks for updating, now good for me.

/approve

**Comment by chendave** [comment]
@oomichi @adtac  would you mind labelling it as "lgtm"? thanks!

**Pull request merged**